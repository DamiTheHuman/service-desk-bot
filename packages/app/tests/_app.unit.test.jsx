import React from 'react';
import {expect, describe, it} from 'vitest';
import {render} from '@testing-library/react';
import App from '../pages/_app';
import Page from '../pages/index';

describe('Index Page', () => {
  it('renders correctly', () => {
    const {asFragment} = render(<App Component={Page} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
