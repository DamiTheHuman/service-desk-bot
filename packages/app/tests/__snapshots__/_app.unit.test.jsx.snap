// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Index Page > renders correctly 1`] = `
<DocumentFragment>
  <header
    class="p-3 bg-dark text-white mb-3"
  >
    <div
      class="container"
    >
      <div
        class="d-flex flex-wrap align-items-center justify-content-center justify-content-lg-start"
      >
        <ul
          class="nav col-12 col-lg-auto me-lg-auto mb-2 justify-content-center mb-md-0"
        >
          <li>
            <a
              class="nav-link px-2 text-secondary"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class="nav-link px-2 text-secondary"
              href="/statistics"
            >
              Statistics
            </a>
          </li>
        </ul>
      </div>
    </div>
  </header>
  <div
    class="container flex-column vh-75 
    p-4 pb-4 overflow-auto"
    id="index"
  >
    <div
      class="d-flex justify-content-between flex-column pb-4"
    >
      <h2>
        Service Desk Bot 
        <span
          class="h4 text-black-50"
        >
          v.1.0.0
        </span>
      </h2>
      <div
        class="message-box"
      >
        <div
          class="message-data d-flex flex-row"
        >
          <div
            class="p-2"
          >
            <i
              class="bi bi-robot fs-4"
            />
          </div>
          <div
            class="p-2"
          >
            <div
              class="container-fluid"
            >
              <div
                class="row row-cols-1"
              >
                <div
                  class="col fw-bold message-data__author-role"
                >
                  Service Desk Bot
                </div>
                <div
                  class="col message-data__message"
                >
                  <div
                    class="messages"
                  >
                    <p
                      class="messages__message"
                    >
                      Say Something to get started!
                    </p>
                  </div>
                  <div
                    class="options"
                  >
                    <button
                      class="options__option btn btn-dark mx-2"
                    >
                      Hi
                    </button>
                    <button
                      class="options__option btn btn-dark mx-2"
                    >
                      Hello
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-auto fixed-bottom container pb-2 bg-white"
      >
        <form
          role="deliver-message"
        >
          <div
            class="input-group"
          >
            <span
              class="input-group-text"
              id="basic-addon1"
            >
              <i
                class="bi bi-person-circle"
              />
            </span>
            <input
              class="message-prompt form-control form-control-lg"
              id="message-prompt"
              name="message"
              placeholder="Message Service Desk Bot..."
              type="text"
              value=""
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
